project_recalculate/models/project_task.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning, ValidationError
+from datetime import datetime
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    from_days = fields.Integer(
+        string='From days',
+        help='Anticipation days from date begin or date end', default=0)
+    estimated_days = fields.Integer(
+        string='Estimated days', help='Estimated days to end', default=1,
+        oldname='anticipation_days')
+
+    @api.one
+    @api.constrains('estimated_days')
+    def _estimated_days_check(self):
+        if not self.estimated_days > 0:
+            raise ValidationError(_('Estimated days must be greater than 0.'))
+
+    def _dates_onchange(self, vals):
```

Comment:
task dates are fields that user can change manually. Is there other way to implement this?